### PR TITLE
string-functions: Explain binary-as-hex behavior

### PR DESCRIPTION
**Pull request by dveeden**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by dveeden** [comment]
For the vale issue see #19118

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -2240,7 +2244,8 @@ The `UNHEX()` function performs the reverse operation of the `HEX()` function. I
 
 > **Note:**
 >
-> The argument must be a valid hexadecimal value that contains `0`–`9`, `A`–`F`, or `a`–`f`. If the argument is `NULL` or falls outside this range, the function returns `NULL`.
+> - The argument must be a valid hexadecimal value that contains `0`–`9`, `A`–`F`, or `a`–`f`. If the argument is `NULL` or falls outside this range, the function returns `NULL`.
+> - In the MySQL Client the [`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) option is enabled by default in interactive mode, this causes the client to display data for which is doesn't know the character set as [`hexadecimal literal`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html) instead. You can use the `--skip-binary-as-hex` option to disable this behavior.
```
Comment (hfxsd) [inline functions-and-operators/string-functions.md:2248]:
```suggestion
> - In the MySQL Client the [`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) option is enabled by default in interactive mode, this causes the client to display data with unknown character set as a [`hexadecimal literal`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html). You can use the `--skip-binary-as-hex` option to disable this behavior.
```

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -638,7 +638,11 @@ The `HEX()` function is used to convert the given argument into a string represe
 - If the argument is a number, `HEX(n)` returns a hexadecimal string representation of `n`. The function treats the argument `n` as a `BIGINT` number, equivalent to using `CONV(n, 10, 16)`.
 - If the argument is `NULL`, the function returns `NULL`.
 
-Examples:
+> **Note:**
+>
+> In the MySQL Client the [`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) option is enabled by default in interactive mode, this causes the client to display data for which is doesn't know the character set as [`hexadecimal literal`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html) instead. You can use the `--skip-binary-as-hex` option to disable this behavior.
```
Comment (hfxsd) [inline functions-and-operators/string-functions.md:643]:
```suggestion
> In the MySQL Client the [`--binary-as-hex`](https://dev.mysql.com/doc/refman/8.0/en/mysql-command-options.html#option_mysql_binary-as-hex) option is enabled by default in interactive mode, this causes the client to display data with unknown character set as a [`hexadecimal literal`](https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html). You can use the `--skip-binary-as-hex` option to disable this behavior.
```

**Review by hfxsd — APPROVED** [review]

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**